### ISSUE TITLE
Can I get older versions (revert to 3.1.5)

### ISSUE THREAD

**Issue by freebrowser1** [utmapp/UTM#4013]
I have installed 3.2.3 but forgot to backup 3.1.5 version and now 3.2.3 does not work properly.
Where can I get 3.1.5 for macOS ?

**Comment by osy** [comment]
Can you please explain what doesn’t work properly? And you can download the older releases by scrolling down.

**Comment by freebrowser1** [comment]
I updated the comment, I already found the older version.
Thanks.

**Comment by conath** [comment]
@freebrowser1 Please provide Debug Log as per [VM Troubleshooting](https://github.com/utmapp/UTM/wiki/Troubleshooting-a-VM#how-to-get-debug-log) guide.

**Comment by freebrowser1** [comment]
> @freebrowser1 Please provide Debug Log as per [VM Troubleshooting](https://github.com/utmapp/UTM/wiki/Troubleshooting-a-VM#how-to-get-debug-log) guide.

I reverted to 3.1.5, so I cannot make a debug log now, I'll try later.
Can I have two UTM installations on the same Mac (i.e. 3.1.5 for production and 3.2.3 (or higher) for testing ) ?

But the problem also occurs on a plain ISO disk image of Centos 7 booted as a UTM VM:

http://isoredirect.centos.org/centos/7/isos/x86_64/CentOS-7-x86_64-Minimal-2009.iso

Got from: https://tecadmin.net/download-centos-7/

**Comment by conath** [comment]
Yes, you can have multiple UTM installed next to each other, they will share the VMs. I do this by renaming the “UTM.app” to “UTM 3.1.5.app” when updating to the new version.

between 3.1.5 and 3.2.x the QEMU backend changed so this is likely why your VM is broken. I need the log to see what backend your VM is set to. Please add the log from running with 3.1.5.

**Comment by osy** [comment]
https://miguelmenendez.pro/en/blog/fast-tsc-calibration-failed/

This source says the error can be ignored?

**Comment by freebrowser1** [comment]
Debug logging of both VM managers:

In 3.2.3 the terminal freezes even with doing a simple `ls -al`. Running the same VM (a clone of the other) under 3.1.5 runs flawlessly.
[debug-centos-utm-315.log](https://github.com/utmapp/UTM/files/8687071/debug-centos-utm-315.log)
[debug-centos-utm-323.log](https://github.com/utmapp/UTM/files/8687072/debug-centos-utm-323.log)

I did not see the 'Fast TSC calibration failed' error anymore when running under 3.2.3, but it is still unusable due to above readons.

**Comment by conath** [comment]
From the log I can see your VM uses the QEMU backend via macOS Intel Hypervisor and the `q35` machine type. 

Potentially the newly introduced QEMU 7 is the reason for your problems. To test this, you could run the same OS in a "vanilla" QEMU 7 install (e. g. through homebrew). The "export QEMU arguments" feature in UTM should get you most of the way there for getting a QEMU VM going. If the issue also occurs while outside UTM, we can report the bug to them.

**Comment by nar3nd3r** [comment]
I am not entirely sure if this related to current issue, but 5.5.7-1-ARCH freshly downloaded from Gallery does break as well, logs attached here.

[boot.log](https://github.com/utmapp/UTM/files/8689502/boot.log)

I can always go ahead and create a new issues, but just wanna be sure, if we are talking separate here.

**Comment by freebrowser1** [comment]
I tried upgrading QEMU under Homebrew but `brew upgrade` did not support QEMU 7 yet. 6.2.0.7 appears to be the latest version. So I wait until QEMU 7 is more mature. It is still a 'zero version' which is sparely tested.

**Comment by osy** [comment]
That’s not true. QEMU’s versioning is literally major first every year and minor throughout the year.

**Comment by osy** [comment]
@freebrowser1 I am doing the follow steps on my Intel Mac

1. Use 3.2.3 wizard to create a Linux VM (GUI, no GL). 64GB HDD, ISO: `CentOS-7-x86_64-Minimal-2009.iso`
2. Start the VM and go through installation, create an admin account `admin`
3. Reboot, eject the ISO, and get login prompt
4. Login as `admin` and type in `ls -al`

On my machine, it works fine (no crash). Am I testing this correctly? Do you have a crash log?

**Comment by osy** [comment]
@nar3nd3r I also tried testing the gallery Arch VM on my Intel Mac (no changes). Logged in and ran `ls -al` no issues. Do you also have a crash log? If so please open a new issue. If it turns out to be the same issue, I will just close it.

**Comment by nar3nd3r** [comment]
@osy how can I get that log for ya, except the one I have up there.

**Comment by osy** [comment]
When you create a new issue you get the following instructions:
> If the app crashed, you need a crash log. To get your crash log, open Console.app, go to `Crash Reports`, and find the latest entry for either QEMU, QEMUHelper, or qemu-\*. Right click and choose `Reveal in Finder`. Attach the report here.

**Comment by nar3nd3r** [comment]
ahh right, I though there are some UTM specific scripts that collects logs, anyways raised https://github.com/utmapp/UTM/issues/4014.

**Comment by freebrowser1** [comment]
> @freebrowser1 I am doing the follow steps on my Intel Mac
> 
>     1. Use 3.2.3 wizard to create a Linux VM (GUI, no GL). 64GB HDD, ISO: `CentOS-7-x86_64-Minimal-2009.iso`
>     2. Start the VM and go through installation, create an admin account `admin`
>     3. Reboot, eject the ISO, and get login prompt
>     4. Login as `admin` and type in `ls -al`

That worked for me as well on my Intel mac. But the VM I use is based on the old BIOS boot scheme, it is actually migrated from a VirtualBox VDI which I converted to qcow2.
I'll try to convert this VM to an EFI boot scheme.

**Comment by freebrowser1** [comment]
Starting Ubuntu 22.04 VM under 3.2.3 has no network. This VM does have an EFI boot.
The Centos 7 VM also has no network under 3.2.3.
It appears network connections do not work properly under QEMU 7 (?).

[debug-ubuntu-2204-utm323.log](https://github.com/utmapp/UTM/files/8692520/debug-ubuntu-2204-utm323.log)

**Comment by freebrowser1** [comment]
Logs attached for Ubuntu 22.04 AR<.
[log-and-pilist.zip](https://github.com/utmapp/UTM/files/8692789/log-and-pilist.zip)

**Comment by osy** [comment]
@freebrowser1 for networking issue, it's probably because the network adapter name has changed. Follow https://mac.getutm.app/gallery/ubuntu-20-04 for ubuntu.

**Comment by osy** [comment]
@freebrowser1 When it gets stuck, can you get a spindump?

1. Open Activity Monitor
2. Find UTM process and double click
3. Click "Sample", copy paste the text to a .txt file, and attach it here.

**Comment by freebrowser1** [comment]
> @freebrowser1 When it gets stuck, can you get a spindump?
> 
>     1. Open Activity Monitor
>     2. Find UTM process and double click
>     3. Click "Sample", copy paste the text to a .txt file, and attach it here.
[spindump.txt](https://github.com/utmapp/UTM/files/8733702/spindump.txt)

**Comment by freebrowser1** [comment]
> @freebrowser1 for networking issue, it's probably because the network adapter name has changed. Follow https://mac.getutm.app/gallery/ubuntu-20-04 for ubuntu.

I did this by folloing the instructions for Ubuntu 20.04:

`ip link show` shows:

````
 1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN mode DEFAULT group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
2: enp0s10: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP mode DEFAULT group default qlen 1000
    link/ether 96:f4:87:c7:d3:64 brd ff:ff:ff:ff:ff:ff
````

and `cat /etc/netplan/00-installer-config.yaml` shows:

````
# This is the network config written by 'subiquity'
network:
  ethernets:
    enp0s10:
      dhcp4: true
  version: 2
 ````
 I did this in UTM 3.1.5 which has a working network.

 But under 3.2.3 the network did not work.
